src/Cms/Auth.php

```diff
@@ -390,6 +390,13 @@ public function login(
 
 		// validate the user and log in to the session
 		$user = $this->validatePassword($email, $password);
+
+		// @codeCoverageIgnoreStart
+		if ($user->id() === 'kirby') {
+			throw new PermissionException('The almighty user "kirby" cannot be used for login, only for raising permissions in code via `$kirby->impersonate()`');
+		}
+		// @codeCoverageIgnoreEnd
```

Comment:
What do you think about moving this check to the `$user->loginPasswordless()` method? There it would fully prevent any kind of login and also we would be able to test this in CI.